packages/runtime-dom/src/apiCustomElement.ts

```diff
@@ -269,11 +269,6 @@ export class VueElement
         this._root = this
       }
     }
-
-    if (!(this._def as ComponentOptions).__asyncLoader) {
-      // for sync component defs we can immediately resolve props
-      this._resolveProps(this._def)
```

Comment:
the following code will overwrite the props value
https://github.com/vuejs/core/blob/961151305a8ca94418956d4aaef2b7617c8875ef/packages/runtime-dom/src/apiCustomElement.ts#L343-L345
so it should be resolved in the connectedCallback